#!/usr/bin/env python
# Created on: Oct 2, 2012
#
#       Author: Pablo Inigo Blasco - Robotics and Computer Architecture (ATC) - University of Seville - 2012
#       License: GPLv3
#http://www.ros.org/wiki/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile

PACKAGE = "rtcus_navigation"
import roslib;roslib.load_manifest(PACKAGE)
from dynamic_reconfigure.parameter_generator import *
gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, "", True)

gen.add("artificial_obstacle_fillup_distance_max", double_t, 0, "Create new artificial obstacles. Specially focused to create an obstacle bound to uncertain areas", 0.1,0.01,100.0)
gen.add("artificial_obstacle_fillup_distance", double_t, 0, "Create new artificial obstacles. Specially focused to create an obstacle bound to uncertain areas", 0.6,0.01,10.0)
gen.add("distance_cutoff", double_t, 0, "(0 meters to deactivate) removes obstacles from the input point cloud to improve the performance publish_obstacle_decoration_info: true #publishes in a topic the created artificial obstacles for representation", 0.0,0.0,10.0)
gen.add("close_back", bool_t, 0, "introduces artificial obstacles closing the laser ranger in de back", True)
gen.add("publish_obstacle_decoration_info", bool_t, 0, "publishes the point cloud of new artificial points", True)

exit(gen.generate(PACKAGE, "unvertainty_obstacles_perception_port", "UncertaintyObstaclesPerceptionPort"))

